warehouse/admin/static/js/warehouse.js

```diff
@@ -185,3 +185,42 @@ if (malwareReportsTable.length) {
   new $.fn.dataTable.Buttons(table, {buttons: ["copy", "csv", "colvis"]});
   table.buttons().container().appendTo($(".col-md-6:eq(0)", table.table().container()));
 }
+
+// Link Checking
+var links = document.querySelectorAll("a");
+links.forEach(function(link){
+  if (link.dataset.checkLinkUrl !== undefined) {
+    var reportLine = {bareUrl: link.href, url: link.dataset.checkLinkUrl, status:0, element : link};
+    fetch(reportLine.url, {
+      method: "GET",
+      mode: "cors",
+    })
+      .then(function(response) {
+        var responseText = "";
+        response.text().then((text) => {
+          responseText = text;
+          console.log(response.status, responseText);
+          if (response.status == 400 && responseText == "Unsupported content-type returned\n") {
```

Comment:
Is 400 the only response status we're looking to handle? What about other non-400s, like a 429 if we get rate limited for some reason? I think 429 is likely to hit the `else` block and not end up in `.catch`

Would we prefer to verify the response earlier with `if (!response.ok) { ...`

Reply:
this is _pretty_ rudimentary and relies on Camo's response of 400 "Unsupported content-type returned" when it successfully loads a page and doesn't find an image. camo won't rate limit us, and i'm not sure that other statuses like 429 would bubble up.

for now, I'd like to just proceed with this and see how it behaves in practice.